src/test/java/org/prebid/server/bidder/screencore/ScreencoreBidderTest.java

```diff
@@ -0,0 +1,203 @@
+import static org.prebid.server.util.HttpUtil.X_FORWARDED_FOR_HEADER;
+import static org.prebid.server.util.HttpUtil.X_OPENRTB_VERSION_HEADER;
+import static org.prebid.server.util.HttpUtil.headers;
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+public class ScreencoreBidderTest extends VertxTest {
+
+    private static final String ENDPOINT_URL = "http://h1.screencore.io/?kp={{AccountId}}&kn={{SourceId}}";
+
+    private ScreencoreBidder target;
+
+    @Before
+    public void setUp() {
+        target = new ScreencoreBidder(ENDPOINT_URL, jacksonMapper);
+    }
+
+    @Test
+    public void creationShouldFailOnInvalidEndpointUrl() {
+        assertThatIllegalArgumentException().isThrownBy(() -> new ScreencoreBidder("invalid_url", jacksonMapper));
+    }
+
+    @Test
+    public void makeHttpRequestsShouldReturnErrorWhenRequestHasInvalidImpression() {
+        // given
+        final ObjectNode invalidExt = mapper.valueToTree(ExtPrebid.of(null, mapper.createArrayNode()));
+        final BidRequest bidRequest = givenBidRequest(impBuilder -> impBuilder.ext(invalidExt));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        assertThat(result.getErrors()).hasSize(1).first()
+                .satisfies(error -> {
+                    assertThat(error.getMessage()).startsWith("Cannot deserialize");
+                    assertThat(error.getType()).isEqualTo(BidderError.Type.bad_input);
+                });
+    }
+
+    @Test
+    public void makeHttpRequestsShouldMakeCorrectRequest() {
+        // given
+        final BidRequest bidRequest = givenBidRequest(
+                identity(),
+                impBuilder -> impBuilder.id("imp_id2").ext(givenImpExt()));
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = target.makeHttpRequests(bidRequest);
+
+        // then
+        final BidRequest expectedBidRequest = givenBidRequest(
+                impBuilder -> impBuilder.ext(null),
+                impBuilder -> impBuilder.id("imp_id2").ext(givenImpExt()));
+        final MultiMap expectedHeaders = headers()
+                .set(CONTENT_TYPE_HEADER, APPLICATION_JSON_CONTENT_TYPE)
+                .set(ACCEPT_HEADER, APPLICATION_JSON_VALUE)
+                .set(X_OPENRTB_VERSION_HEADER, "2.5")
+                .set(USER_AGENT_HEADER, "ua")
+                .set(X_FORWARDED_FOR_HEADER, List.of("ipv6", "ip"));
+        final Result<List<HttpRequest<BidRequest>>> expectedResult = Result.withValue(HttpRequest.<BidRequest>builder()
+                .method(HttpMethod.POST)
+                .uri("http://h1.screencore.io/?kp=accountId&kn=placementId")
+                .headers(expectedHeaders)
+                .impIds(Set.of("imp_id", "imp_id2"))
+                .body(jacksonMapper.encodeToBytes(expectedBidRequest))
+                .payload(expectedBidRequest)
+                .build()
```

Comment:
Let's separate this test to three other
- headers check
- body check
- impIds check

Reply:
I believe currently the test is good in terms of comparing the whole object that is built as a result of the method call... basically what you suggest is splitting a single test case into three and compare different parts of the same object

I prefer keeping this test as is, still it's a main happy flow test, but if you have some other specific test cases for the happy flow, let me know